packages/vue-3/src/EditorContent.ts

```diff
@@ -78,18 +79,6 @@ export const EditorContent = defineComponent({
 
       editor.contentComponent = null
       editor.appContext = null
-
-      if (!editor.options.element.firstChild) {
-        return
-      }
-
-      const newElement = document.createElement('div')
-
-      newElement.append(...editor.options.element.childNodes)
```

Comment:
This code does seem strange to me and unfortunately does not look like it was ever really documented. My best guess is that node views need to be unmounted as well, and to do that they still need to be part of the DOM so they are being copied over to avoid an error with them being unmounted after the view is destroyed.

So I think that the best way to test this is to have a vue app that is utilizing node views and see if it still can be unmounted properly. If it cannot, then we will need to find another way around this.

The first thought that comes to mind would be to force the editor to destroy the node views first, then to destroy the editor view. But, let's see if we can get away with just this simple solution first and discuss afterwards

Reply:
This part of the code comes from the `beforeDestroy` of vue 2 `EditorContent`;that's why I removed it as it may be "outdated".

I also extended the example to include a custom component within the editor (and I needed to remove the part saying `// destroy nodeviews before vue removes dom element` for it to work properly).

For both examples, I made sure that all the nodes were removed from the DOM.

If needed, the example now includes a more complete example with a `NodeViewWrapper` component within the editor 🙂